### PR TITLE
[MRG] Fix docs and add Test Python package

### LINKED ISSUE #928: bids-validator in Python for Windows OS
Hello,

I am working on Windows OS and I had difficulty using bids-validator python package. 
bids-validator couldn't match my path with the rules. I changed the separator of my path '\\' in '/'  and it seems working now.

I just don't know if it's the best solution. Do you know a better solution?

Thank you
Aude

### PR DESCRIPTION
**Pull request by sappelhoff**
closes #928 

related to https://github.com/bids-standard/bids-validator/issues/928#issuecomment-603968578

- fix docstrings and style of Python package
- add test script for use with pytest
- add test run to circleCI

Possible To Do (not this PR):
- run tests on Windows and MacOS

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+      - image: circleci/python:stretch
+    steps:
+      - checkout
+      - run: pip install --user pytest flake8 pydocstyle
+      - run: pwd
+      - run: ls bids-validator
+      - run: ls bids-validator/bids-validator
+      - run: pytest bids-validator/bids-validator/test_bids_validator.py
+      - run: flake8 bids-validator/bids-validator/bids-validator.py
+      - run: pydocstyle bids-validator/bids-validator/bids-validator.py
```
Comment (effigies) [inline .circleci/config.yml]:
```suggestion
      - run: pytest bids-validator/bids_validator
      - run: flake8 bids-validator/bids_validator
      - run: pydocstyle bids-validator/bids_validator/bids_validator.py
```

Comment (sappelhoff) [reply .circleci/config.yml]: gosh ... thanks. `-` versus `_`. The room I am sitting in is too dark :man_facepalming:

**Comment by sappelhoff** [comment]
Regarding testing on Windows, I find Azure Pipelines interesting. In this article it seems that it's relatively straight forward to set up tests for Windows, MacOS and Linux all in one go:

https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started-multiplatform?view=azure-devops

We would gain MacOS tests ... and have the advantage of running Windows tests on GitHub instead of on GitLab :-)

**Comment by nellh** [comment]
> Regarding testing on Windows, I find Azure Pipelines interesting. In this article it seems that it's relatively straight forward to set up tests for Windows, MacOS and Linux all in one go:
> 
> https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started-multiplatform?view=azure-devops
> 
> We would gain MacOS tests ... and have the advantage of running Windows tests on GitHub instead of on GitLab :-)

GitLab is close to releasing free macOS runners and builds for GitHub forks, which would make them pretty equivalent.

**Comment by sappelhoff** [comment]
> GitLab is close to releasing free macOS runners and builds for GitHub forks, which would make them pretty equivalent.

okay, sounds cool. thanks for the info @nellh. I think we can merge this PR then and wait for the GitLab pipelines to become more usable from GitHub ... and then add additional OSses to this test.

**Comment by sappelhoff** [comment]
but talking about this ... it seems that "GitHub actions" now also permit open source repos to perform CI jobs ... on MacOS, Linux, Windows :thinking: 

https://github.com/features/actions

if we can do that on GitHub, I think it'd be better than at GitLab

EDIT: I have implemented GitHub actions in a repo here: https://github.com/sappelhoff/pyprep/blob/master/.github/workflows/python_tests.yml

It's very straight forward, at least for the simple workflow that I required.

File: bids-validator/bids_validator/test_bids_validator.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+]
+
+VALIDATOR = BIDSValidator()
+
+
+@pytest.mark.parametrize('data', DATA)
+def test_BIDSValidator(data):
+    """Test basic functionality of BIDSValidator class."""
+    fpath, result = data
+    assert VALIDATOR.is_bids(fpath) == result
```
Comment (effigies) [inline bids-validator/bids_validator/test_bids_validator.py]: This looks the same as the doctest. Just use `pytest --doctest-modules` to run those tests. No need to duplicate as separate tests.

**Comment by rwblair** [comment]
Be curious to see how far we get under the free plan with the macos and windows multipliers:

https://help.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions

@sappelhoff is your account for pyprep free? have there been any limits issues?

**Comment by sappelhoff** [comment]
@rwblair as far as I can see it, GitHub Actions are free of charge (and no limits) for public repos:

> **GitHub Actions usage is free for public repositories.** For private repositories, each GitHub account receives a certain amount of free minutes and storage, depending on the product used with the account.

(from your link, emphasis added)

**Pull request merged**